client/src/components/Module.jsx

```diff
@@ -2,50 +2,76 @@ import { React, useState, useEffect } from 'react';
 import PropTypes, { string } from 'prop-types';
 import { Link } from 'react-router-dom';
 import {
-  ref, getStorage, getDownloadURL,
+  ref, getStorage, getDownloadURL, getMetadata,
 } from 'firebase/storage';
 import styles from '../styles/Modules.module.css';
 
 function Module(props) {
   const {
-    title, body, attachments, child, link,
+    title, body, child, links,
   } = props;
 
-  const [img, setImg] = useState('');
-  console.log(img);
+  const [files, setFiles] = useState([]);
 
-  const setImageURL = (imgLink) => {
-    if (imgLink.length > 0) {
-      const storage = getStorage();
-      const spaceRef = ref(storage, imgLink);
-      getDownloadURL(spaceRef).then((url) => {
-        setImg(url);
-      }).catch((error) => {
-        console.error(error.message);
-      });
-    } else {
-      setImg('');
+  const updateImageURL = async (fileLinks) => { // i'm gonna be slow bc i contain a Promise function!
+    setFiles([]);
+    const fileContents = [];
+    if (fileLinks.length > 0) {
+      await Promise.all(fileLinks.map(async (fileLink) => { // i'm also gonna be slow bc i contain a Promise function!
+        const storage = getStorage();
+        const spaceRef = ref(storage, fileLink);
+        const file = await getMetadata(spaceRef); // getMetadata is a promise function,
+        // ^ so it's very slow (you need to wait for it to be fulfilled before u do anything w it)
+        const fileType = file.contentType;
+        const url = await getDownloadURL(spaceRef);
+        fileContents.push({ url, fileType });
+      }));
+      setFiles(fileContents);
     }
   };
 
-  useEffect(() => { setImageURL(link); }, [link]);
+  useEffect(() => { updateImageURL(links); }, [links]);
 
   return (
     <div>
       <div className={styles.title}>{title}</div>
       <div className={styles.body}>{body}</div>
-      <div className={styles.attachments}>{attachments}</div>
-      {/* keep bottom code for reference */}
-      {/* <div>
-        {link.map((image) => {
+      {/* checks if file is img (png, jpg, jpeg), vid (np4, mpeg, mov), or pdf */}
+      <div>
+        {files.map((file) => {
+          if (file.fileType === 'image/png' || file.fileType === 'image/jpeg') {
             return (
-              <div key={image} className="image">
-                  <img src={image} alt="" width="40%" height="auto" />
+              <div>
```

Comment:
What do u mean about "i" here? I think you can even use the url as a key if needed, for all the divs inside map functions.